src/index.js

```diff
@@ -242,9 +242,12 @@ export function renderer(view, state, actions) {
         out += frame.footer
         stack.pop()
       } else {
-        const node = resolveNode(frame.children[frame.childIndex++], state, actions)
+        let node = resolveNode(frame.children[frame.childIndex++], state, actions)
 
         if (node != null && typeof node !== 'boolean') {
+          if (node.type === 2 && node.lazy) {
+            node = node.lazy.view(node.lazy)
+          }
```

Comment:
Shouldn't this part be inside `resolveNode` function so it will resolve any `Lazy` components recursively?
Would be cool to have a test for nested Lazy.

Reply:
Good catch! This  place didn't feel right to me. It's so much cleaner in the resolveNode function. Will send an updated PR in a sec.

Reply:
Totally agree on the test for the nested lazy! I will add it.

Reply:
I've just checked and interestingly HA2 doesn't support nested lazy. But since it doesn't cost us anything to add it here I'd leave the nested lazy support.

Reply:
Actually there was a problem not just with nested `Lazy({ render: Lazy({...}) })` but also with using functional components inside Lazy view `Lazy({ render: Component })`.
I think we are good now. I will create a new release later today. Thank you!